web/static/js/thesis-editor.js

```diff
@@ -203,6 +269,14 @@ class ThesisEditor extends React.Component {
     } else {
       el.classList.remove('thesis-tray-open')
     }
+
+    document.title = this.state.title
+    this.pageDescriptionMetaTag().content = this.state.description
+    thesisContainer.setAttribute('data-redirect-url', this.state.redirectURL)
```

Comment:
This is part of the `componentDidUpdate`, and reaches into the DOM to update the non-React parts of Thesis to reflect the new state.